src/txmempool.cpp

```diff
@@ -705,6 +705,7 @@ void CTxMemPool::removeForBlock(const std::vector<CTransactionRef>& vtx, unsigne
 void CTxMemPool::_clear()
 {
     mapTx.clear();
+    vTxHashes.clear();
```

Comment:
Just for the sake of not having invalid iterators even temporarily, I prefer this before mapTx is cleared.

Reply:
Updated to move vTxHashes clear before mapTx clear.